docs/reference-spec.md

```diff
@@ -0,0 +1,64 @@
+# Reference and Naming Specification
+
+This section of the Bindle specification defines how naming and referencing works.
+
+## Bindle Name (and Version)
+
+A bindle has a name. For example, if we were to write a `hello world` bindle, it may have the name `hello_world`.
+
+When a bindle is concretized into an invoice, it MUST have a name and a version. Version MUST be a valid [https://semver.org](SemVer 2 version).
+
+A bindle name can be one or more characters from the set of unicode alphabetic and numeric characters, as well as the characters `_` and `/`.
+
+TODO: Are there other characters we should allow? `-` causes problems with algorithms that autoconvert between `_` and `-`, because of ambiguity. `+` and `%` are legal in URLs, and % might be useful as an escape character, per URLs. A bindle name MUST NOT start with the string `bindle:` (the `:` is not a legal character in bindle names).
+
+Example bindle names:
+
+- `a`
+- `1_b`
+- `käse`
+- `example.com/hello/world`
+
+Bindle creators SHOULD use a domain syntax for naming, where the domain name is associated with the organization of the creators. Examples:
+
+- `example.com/hello_world`
+- `github.com/deislabs/example_bindle`
+
+## Invoice Name
+
+An invoice name is the concatenation of the bindle name and version. The `/` character is used to combine these two pieces of information.
+
+Examples:
+
+- `example.com/hello_world/1.2.3`
+- `github.com/deislabs/example_bindle/123.234.34567-alpha.9999+hellothere`
```

Comment:
![tenor-193977382](https://user-images.githubusercontent.com/4294228/105741695-d9dcee80-5ef7-11eb-866e-95e0c85e3919.gif)

Reply:
You know, for those people who cut a new release on every single git push